src/plugins/discover/public/embeddable/saved_search_grid.tsx

```diff
@@ -18,14 +18,13 @@ export interface DiscoverGridEmbeddableProps extends DiscoverGridProps {
   totalHitCount: number;
 }
 
-export const DataGridMemoized = React.memo((props: DiscoverGridProps) => (
-  <DiscoverGrid {...props} />
-));
+export const DataGridMemoized = memo(DiscoverGrid);
 
 export function DiscoverGridEmbeddable(props: DiscoverGridEmbeddableProps) {
-  const services = getServices();
   const [expandedDoc, setExpandedDoc] = useState<ElasticSearchHit | undefined>(undefined);
 
+  const services = useMemo(() => getServices(), []);
```

Comment:
tiny nit, since you break down everything here soon ... there should not be a need to use 'useMemo' here, since `services` shouldn't change and only a reference is returned by `getServices()`